service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManagerOptions.java

```diff
@@ -0,0 +1,17 @@
+package com.microsoft.azure.sdk.iot.service;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Configurable options for all registry manager operations
+ */
+public class RegistryManagerOptions
+{
+    /**
+     * The options that specify what proxy to tunnel through. If null, no proxy will be used
+     */
+    @Getter
+    @Setter
+    private ProxyOptions proxyOptions;
```

Comment:
Why not add `@AllArgsConstructor` for 1 line initialization.

Reply:
I don't want this to be construed as an essential option. As such, there should be a constructor that doesn't take this

Reply:
For instance, if we ever add something like HTTP readTimeout/connectTimeout, I wouldn't want customers to need a proxy here

Reply:
yeah, good point.

Reply:
But I still feel the options object in itself should not be null, but have a default value

Reply:
I'm not sure I see the value in holding a non-null default proxyOptions class. Each of those null checks would just turn into checks on if the proxy is of type DIRECT (no proxy) or type HTTP.

Reply:
For AMQP in particular, it isn't just a straight pass through. We have to add a transport layer manually if there is a proxy to go through

Reply:
What about a builder pattern here instead? Since all of the hypothetical fields in this class are optional, and we don't know how many options there may be in the future, a builder could fit in well here

Reply:
Suppose tomorrow we add a parameter for Http connect timeout, which has a default value of 1min, but settable. So now, the `RegistryManagerOptions` class is always non-null (irrespective of proxy). 
So something like this, which we have below:
`options != null ? options.getProxyOptions() : null`
would want to get proxy options, even though we did not set it/ want to use it.

Reply:
Yeah, a builder would work here. We can set @NonNull for required params

Reply:
Okay, I pushed a commit that makes these two options classes builder based

Reply:
I can make the RegistryManagerOptions and ServiceClientOptions always non-null, for simplicity, sure. The default proxyOptions will still be null, though

Reply:
Yes, of course, we don't want to force/ have a default for proxy